tests/backends/build_system/unit/test_install.py

```diff
@@ -0,0 +1,201 @@
+                os.path.join("lib_dir", "aws_completer"),
+                os.path.join("bin_dir", "aws_completer"),
+            ),
+        ]
+
+    @if_windows
+    def test_install_exe_windows(self):
+        utils = FakeUtils(is_exe=True)
+        installer = Installer("build_dir", utils=utils)
+        installer.install("lib_dir", "bin_dir")
+
+        assert utils.calls == [
+            ("isdir", os.path.join("build_dir", "exe")),
+            ("isdir", "lib_dir"),
+            (
+                "copy_directory",
+                os.path.join("build_dir", "exe", "aws", "dist"),
+                "lib_dir",
+            ),
+            ("path_exists", "bin_dir"),
+            ("makedirs", "bin_dir"),
+            ("islink", os.path.join("bin_dir", "aws.cmd")),
+            (
+                "symlink",
+                os.path.join("lib_dir", "aws.cmd"),
+                os.path.join("bin_dir", "aws.cmd"),
+            ),
+        ]
+
+    @skip_if_windows
+    def test_install_venv(self):
+        utils = FakeUtils(is_exe=False)
+        installer = Installer("build_dir", utils=utils)
+        installer.install("lib_dir", "bin_dir")
+
+        assert utils.calls == [
+            ("isdir", "build_dir/exe"),
+            ("isdir", "lib_dir"),
+            ("copy_directory", "build_dir/venv", "lib_dir"),
+            ("read_file_lines", "lib_dir/bin/aws"),
+            ("write_file", "lib_dir/bin/aws", "#!lib_dir/bin/python\n"),
+            ("read_file_lines", "lib_dir/bin/aws_completer"),
+            (
+                "write_file",
+                "lib_dir/bin/aws_completer",
+                "#!lib_dir/bin/python\n",
+            ),
+            ("path_exists", "bin_dir"),
+            ("makedirs", "bin_dir"),
+            ("islink", "bin_dir/aws"),
+            ("symlink", "lib_dir/bin/aws", "bin_dir/aws"),
+            ("islink", "bin_dir/aws_completer"),
+            ("symlink", "lib_dir/bin/aws_completer", "bin_dir/aws_completer"),
+        ]
+
+    @if_windows
+    def test_install_venv_windows(self):
+        utils = FakeUtils(is_exe=False)
+        installer = Installer("build_dir", utils=utils)
+        installer.install("lib_dir", "bin_dir")
+
+        assert utils.calls == [
+            ("isdir", "build_dir\\exe"),
+            ("isdir", "lib_dir"),
+            ("copy_directory", "build_dir\\venv", "lib_dir"),
+            ("read_file_lines", "lib_dir\\Scripts\\aws.cmd"),
+            (
+                "write_file",
+                "lib_dir\\Scripts\\aws.cmd",
+                '@echo off & "lib_dir\\Scripts\\python.exe" -x "%~f0" %* & goto :eof\n',
+            ),
+            ("path_exists", "bin_dir"),
+            ("makedirs", "bin_dir"),
+            ("islink", "bin_dir\\aws.cmd"),
+            ("symlink", "lib_dir\\Scripts\\aws.cmd", "bin_dir\\aws.cmd"),
+        ]
+
+
+class TestUninstaller:
+    @pytest.mark.skipif(
+        platform.system() == "Windows", reason="Non-windows test"
+    )
```

Comment:
Is there a particular reason we are switching the style for skip ifs here? Before we were using the decorator `if_windows` and `skip_if_windows`